tests/ignite/metrics/test_precision.py

```diff
@@ -345,58 +357,40 @@ def get_test_cases():
             _test(y_pred, y, batch_size)
 
 
-def test_incorrect_type():
+@pytest.mark.parametrize("average", [False, True, "micro", "weighted"])
+def test_incorrect_type(average):
     # Tests changing of type during training
 
-    def _test(average):
-        pr = Precision(average=average)
-        assert pr._updated is False
-
-        y_pred = torch.softmax(torch.rand(4, 4), dim=1)
-        y = torch.ones(4).long()
-        pr.update((y_pred, y))
-        assert pr._updated is True
-
-        y_pred = torch.randint(0, 2, size=(4,))
-        y = torch.ones(4).long()
-
-        with pytest.raises(RuntimeError):
-            pr.update((y_pred, y))
+    pr = Precision(average=average)
+    assert pr._updated is False
 
-        assert pr._updated is True
+    y_pred = torch.softmax(torch.rand(4, 4), dim=1)
+    y = torch.ones(4).long()
+    pr.update((y_pred, y))
+    assert pr._updated is True
 
-    _test(average=True)
-    _test(average=False)
+    y_pred = torch.randint(0, 2, size=(4,))
+    y = torch.ones(4).long()
 
-    pr1 = Precision(is_multilabel=True, average=True)
-    pr2 = Precision(is_multilabel=True, average=False)
-    assert pr1._updated is False
-    assert pr2._updated is False
-    y_pred = torch.randint(0, 2, size=(10, 4, 20, 23))
-    y = torch.randint(0, 2, size=(10, 4, 20, 23)).long()
-    pr1.update((y_pred, y))
-    pr2.update((y_pred, y))
-    assert pr1._updated is True
-    assert pr2._updated is True
-    assert pr1.compute() == pytest.approx(pr2.compute().mean().item())
+    with pytest.raises(RuntimeError):
+        pr.update((y_pred, y))
 
+    assert pr._updated is True
 
-def test_incorrect_y_classes():
-    def _test(average):
-        pr = Precision(average=average)
 
-        assert pr._updated is False
+@pytest.mark.parametrize("average", [False, True, "micro", "weighted"])
+def test_incorrect_y_classes(average):
+    pr = Precision(average=average)
 
-        y_pred = torch.randint(0, 2, size=(10, 4)).float()
-        y = torch.randint(4, 5, size=(10,)).long()
+    assert pr._updated is False
 
-        with pytest.raises(ValueError):
-            pr.update((y_pred, y))
+    y_pred = torch.randint(0, 2, size=(10, 4)).float()
+    y = torch.randint(4, 5, size=(10,)).long()
 
-        assert pr._updated is False
+    with pytest.raises(ValueError):
+        pr.update((y_pred, y))
 
-    _test(average=True)
-    _test(average=False)
+    assert pr._updated is False
 
 
 def _test_distrib_integration_multiclass(device):
```

Comment:
Is there any reason that this test has not been parametrized?

Reply:
This test and other similar ones are used in test_distrib_hvd, test_distrib_gloo_cpu_or_gpu, test_distrib_nccl_gpu tests. IMO, it is not obvious how to properly parametrize them. However, there was a task to rewrite distributed tests such that we do not need to write all: `test_distrib_hvd, test_distrib_gloo_cpu_or_gpu, test_distrib_nccl_gpu` but only `_test_distrib_integration_multiclass` and they could be executed with hvd, gloo, nccl confs...